elastic/eui
# [EuiDataGrid] Add new beta `rowHeightsOptions.autoBelowLineCount` feature flag

## Summary

closes https://github.com/elastic/eui/issues/7835

As always, I recommend code reviewing [by commit](https://github.com/elastic/eui/pull/8096/commits).

> [!warning]
> This is a **beta** feature added for Discover testing, and is and may contain performance issues or other bugs (see discussion in #7835 and various bug reports around auto height grids). This beta status can be removed if Discover is able to use this feature extensively in production without major user issues for an extended period of time.

| Before | After |
|--------|--------|
| <img width="1017" alt="" src="https://github.com/user-attachments/assets/87d1919b-b8d8-482d-82d2-923124ff15a2"> | <img width="1015" alt="" src="https://github.com/user-attachments/assets/f4a3ecfe-1bba-4aa3-81b4-4790433cdc00"> | 

## QA

- Go to https://eui.elastic.co/pr_8096/storybook/?path=/story/tabular-content-euidatagrid-rowheightsoptions-prop--auto-below-line-count
- Open the display settings popover in the datagrid toolbar
- [x] Confirm the `Lines per row` control shows "Max" instead of "[Static](https://eui.elastic.co/pr_8096/storybook/?path=/story/tabular-content-euidatagrid-rowheightsoptions-prop--line-count)"
- [x] Confirm that the single sentence rows render at a single line/row height, and the longer sentences cap out at the line count specified (increase and decrease this number to test)
- [x] Confirm that switching to `Auto` height causes the full lorem ipsum text to be displayed
- [x] Toggle the `autoBelowLineCount` flag in the Storybook controls to compare this behavior, and confirm the `false` behavior acts as before/production

### General checklist

- Browser QA
    ~- [ ] Checked in both **light and dark** modes~
    ~- [ ] Checked for **accessibility** including keyboard-only and screenreader modes~
- Docs site QA
    - ~[ ] Added **[documentation](https://github.com/elastic/eui/blob/main/wiki/contributing-to-eui/documenting)**~ Skipping documentation for now while feature is in beta
- Code quality checklist
- Release checklist
    ~- [ ] If applicable, added the **breaking change** issue label (and filled out the breaking change checklist)~
- Designer checklist - N/A
